pyanaconda/ui/lib/subscription.py

```diff
@@ -267,20 +268,29 @@ def register_and_subscribe(payload, progress_callback=None, error_callback=None)
     task_proxy = SUBSCRIPTION.get_proxy(task_path)
     task.sync_run_task(task_proxy)
 
-    # report attaching subscription was successful
+    # check if the current installation source should be overridden by
+    # the CDN source we can now use
+    # - at the moment this is true only for the CDROM source
+    source_proxy = payload.get_source_proxy()
+    if source_proxy.Type in SOURCE_TYPES_OVERRIDEN_BY_CDN:
```

Comment:
Why?

Reply:
Back during 8.2 the conclusion was reached that RHEL DVD images should default to sing the CDN installation source by default as by the time customers actually get their hands on the image, there will be many updates, many of them security sensitive. And over time more and more packages on the DVD will be outdated while the packages from the CDN will be always up to date.

By defaulting to installation from the CDN even on the DVD installation media the resulting system will be secure and the customer can also avoid installing updates right on the fist start after the installation, resulting in better experience.

To implement this, we need to track if the currently set source is one of those that can be overridden and then also roll it back on un-registration to maintain consistency.

Reply:
It is just a very confusing implementation.